Fix failing E2E tests #166881191

akegan opened the pull request.

akegan committed:
[#166881191] Improve new sale app E2E test

Make lending agent configurable, add assertions to make failures clearer. Add error checking to non-sale apps too.

akegan committed:
[#166881191] WIP try to catch substatus save errors

akegan commented on spec/javascript/support/puppeteer/steps/supplementalApplicationSteps.js:
```diff
@@ -8,12 +10,32 @@ const testStatusModalUpdate = async (page) => {
+  const errors = await page.$$eval('span.error', divs => divs.map(d => d.textContent))
+  expect(errors).toStrictEqual([])
+  // Verify that no salesforce save errors present on save
+  try {
```
@LoganArnett I'm having a hard time getting this to work. 

My goal is to have a useful error that pops up if there's an issue saving substatuses, ideally with the specific status and substatus that's been sent logged. 

The issue is that asserting that something isn't present while allowing for some time to get a response isn't working great

LoganArnett replied:
I am not sure if this is the best way to handle this. Another option would be to use another add-on that can actually listen to `http` requests. Something like Sinon gives you the ability to do something like:
```
page.on('response', response => {
    const endpoint = response.url().split('/').pop();
    if (endpoint === "/update_status") {
        const responseBody = await response.json();
        console.log(responseBody)
        // Here we can decipher whether it was an error or not form the API side
    }
})

LoganArnett replied:
This is the simpler smaller option that is available through Puppeteer but Sinon would allow us to mock and digest more API requests if we see a need for it

akegan replied:
@LoganArnett what's the simpler smaller option?

akegan committed:
[#166881191] Try catching status modal save salesforce errors

Add an expect statement and logging around failed field update comment requests

akegan commented:
This is passing both on semaphore-full and semaphore-prod now, so it's ready for review!

LoganArnett commented on spec/javascript/e2e/ApplicationNewPage.e2e.js:
```diff
@@ -29,6 +29,9 @@ describe('ApplicationNewPage', () => {
 
     // Save the application
     await page.click('.save-btn')
+    // Verify that no errors are present on save
```
Everything looks good but this block of code is in here a number of times so maybe we should make a simple helper function even just at the top of this file so we do not have the duplicate blocks throughout

akegan committed:
[#166881191] Create function for getting form errors

Try to reduce code redundancy

akegan commented:
@LoganArnett ready for re-review

LoganArnett reviewed (approved):
Looks good to me
